weather.js

```diff
@@ -113,425 +80,140 @@ function tempToCel(location) {
                 if (response.ok) {
                     console.log(response);
                     return response.json();
-                }
+                } 
             })
             .then(function (result) {
-                jsonFile = result;
-                console.log(jsonFile);
-                return jsonFile;
-            })
-            .then((data) => {
-                var resultObj1 = data;
-                var max_temp = resultObj1.main.temp_max;
-                var min_temp = resultObj1.main.temp_max;
-                var current_temp = resultObj1.main.temp;
-
-                hightemp.innerText = Math.round((max_temp - 32) * 5 / 9) + '℃';
-                lowtemp.innerText = Math.round((min_temp - 32) * 5 / 9) + '℃';
-                currenttemp.innerText = Math.round((current_temp - 32) * 5 / 9) + '℃';
+                //jsonFile = result;
+                console.log(result);
+                return result;
             })
+            .catch(function (error) {
+                console.log(error);
+            });
     }
+    return jsonFile;
```

Comment:
There is a dependency here that is not easy to catch.

1. Open the main page
2. Type in a zip code
3. Press Celsius or Fahrenheit
4. Error appears in console.

This is due to the expectation that the weather data is pulled via the **Search** button first.

There are 3 solutions that come to mind:

1. Requires Search to be pressed first and only show/hide C/F buttons if there is a valid result.
    * Taking this approach means that you don't actually have to `fetch` for the conversion methods and just convert the `jsonFile` data.  
2. Handle the result of a promise in both `convertToFah` and `covertToCel` functions.
3. Promote `convertToFah` and `covertToCel` to `async` functions and use `await` on the `fetchWeatherData` call but still return the `Promise`.
    * [MDN - Making asynchronous programming easier with `async` and `await`](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await)

Reply:
Sure David, will work on this.

Reply:
Hi David, I choose option 1 and added a solution to fix the C/F buttons. Now these buttons will not be visible only after a valid result comes back after clicking 'Search' button. But i guess i have not completely satisfied what u have mentioned in option 1 regarding not to have to fetch for conversion methods.

https://github.com/DeepthiSF/WhatsTheWeatherToday/commit/56864098cb29dce647e855cd98ed596e6a7fe59d

Reply:
I see. You've still got the `fetchWeatherData()` call in `covertTo*`. As stated, these functions really just need the `jsonFile` variable to be available. Try setting `const resultObj1 = jsonFile` instead of the function call.

Reply:
Made the suggested changes.

https://github.com/DeepthiSF/WhatsTheWeatherToday/commit/bed338d372282af751ca574de5dae8dd3cc65c52